service/src/main/java/bio/terra/cbas/controllers/RunSetsApiController.java

```diff
@@ -82,107 +126,147 @@ public ResponseEntity<RunSetStateResponse> postRunSet(RunSetRequest request) {
       methodDao.createMethod(method);
     } catch (JsonProcessingException e) {
       log.warn("Failed to record method to database", e);
-      return new ResponseEntity<>(HttpStatus.INTERNAL_SERVER_ERROR);
+      return new ResponseEntity<>(
+          new RunSetStateResponse()
+              .errors("Failed to record method to database. Error(s): " + e.getMessage()),
+          HttpStatus.INTERNAL_SERVER_ERROR);
     }
 
-    // Create a new run_set:
+    // Create a new run_set
     UUID runSetId = UUID.randomUUID();
     RunSet runSet = new RunSet(runSetId, method);
     runSetDao.createRunSet(runSet);
 
-    // Fetch the record from WDS:
-    RecordResponse recordResponse;
-    List<String> recordIds;
-    try {
-      String recordType = request.getWdsRecords().getRecordType();
-      recordIds = request.getWdsRecords().getRecordIds();
-      recordRecordsInRequest(recordIds.size());
-      recordResponse = wdsService.getRecord(recordType, recordIds.get(0));
-    } catch (ApiException e) {
-      log.warn("Record lookup failed. ApiException", e);
-      // In lieu of doing something smarter, forward on the error code from WDS:
-      return new ResponseEntity<>(HttpStatus.valueOf(e.getCode()));
-    }
+    // For each Record ID, build workflow inputs and submit the workflow to Cromwell
+    List<RunStateResponse> runStateResponseList =
+        buildInputsAndSubmitRun(request, runSet, recordResponses);
 
-    // Build the inputs set from workflow parameter definitions and the fetched record:
-    Map<String, Object> params =
-        InputGenerator.buildInputs(request.getWorkflowInputDefinitions(), recordResponse);
+    // Figure out how many runs are in Failed state. If all Runs are in an Error state then mark the
+    // Run Set as Failed
+    RunSetState runSetState;
+    List<RunStateResponse> runsInErrorState =
+        runStateResponseList.stream()
+            .filter(run -> CbasRunStatus.fromValue(run.getState()).equals(SYSTEM_ERROR))
```

Comment:
There's also the other ERROR type we should probably include too

Reply:
Good point. I can add that 👍

Reply:
Added the other error state in this check.